### PR TITLE
ref: return multi error

### PR DESCRIPTION
**Pull request by s4kh**
#### Summary
Returns multi error instead of printing and returning nil

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21211

### CODE REVIEW

**Comment by s4kh** [comment]
@isacikgoz when we use `multierror.Append` -> the incoming error is the following:
```
actual  : "1 error occurred:\n\t* unable to find channel 'teamId:channel1'\n\n"
```
Isn't this kind of getting hard to test?

File: commands/channel.go [hunk 1]
```diff
@@ -597,17 +598,19 @@ func deleteChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) erro
...
 	channels := getChannelsFromChannelArgs(c, args)
 	for i, channel := range channels {
 		if channel == nil {
-			printer.PrintError("Unable to find channel '" + args[i] + "'")
+			result = multierror.Append(result, fmt.Errorf("unable to find channel '%s'", args[i]))
 			continue
 		}
 		if _, err := c.PermanentDeleteChannel(channel.Id); err != nil {
-			printer.PrintError("Unable to delete channel '" + channel.Name + "' error: " + err.Error())
+			result = multierror.Append(result, fmt.Errorf("unable to delete channel '%s' error: %v", channel.Name, err))
```
Comment (isacikgoz) [inline commands/channel.go:610]:
```suggestion
			result = multierror.Append(result, fmt.Errorf("unable to delete channel %q: %w", channel.Name, err))
```

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Hey @s4kh this looks good to me in general. Thanks for the contribution!! Could you address my comments?

**Comment by s4kh** [comment]
> Hey @s4kh this looks good to me in general. Thanks for the contribution!! Could you address my comments?

@isacikgoz Addressed the requested changes.

File: commands/channel_test.go [hunk 2]
```diff
@@ -2862,8 +2864,10 @@ func (s *MmctlUnitTestSuite) TestDeleteChannelsCmd() {
 
 		arg := teamName + ":" + channelName
 		err := deleteChannelsCmdF(s.client, cmd, []string{arg})
-		s.Require().Nil(err)
-		s.Require().Equal(fmt.Sprintf("Unable to find channel '%s'", arg), printer.GetErrorLines()[0])
+		// s.Require().EqualError(err, fmt.Sprintf("unable to find channel '%s'", arg))
```
Comment (isacikgoz) [inline commands/channel_test.go:2867]: Can you remove the commented out code?

**Review by isacikgoz — COMMENTED** [review]
Looks good, just a small nit, it's good to go after that. Thanks @s4kh

**Review by isacikgoz — APPROVED** [review]
Thanks for the contribution @s4kh Looks good to me!

**Review by noxer — APPROVED** [review]

**Comment by noxer** [comment]
Ran the test manually, all good.

**Pull request merged**